docs/safe-haven-services/open-ondemand/containers.md

```diff
@@ -0,0 +1,109 @@
+
+| Container Registry | URL prefix | Example  |
+| ------------------ | ---------- | ------- |
+| DockerHub | `ghcr.io/` | `ghcr.io/mikej888/hello-tre:1.0` |
+
+---
+
+## Sharing files between a back-end and a container
+
+When the container is run, three directories on the back-end are mounted into the container:
+
+| Back-end directory | Container directory | Description |
+| ------------------ | ------------------- | ----------- |
+| Project-specific `/safe_data/` subdirectory | `$HOME/safe_data/` OR `/safe_data/PROJECT_SUBDIRECTORY/`| If `$HOME/safe_data/` exists in your home directory on the back-end, then that is mounted. Otherwise, a subdirectory of `/safe_data/` corresponding to your project (and inferred from your user group) is mounted, if such a subdirectory can be found. |
+| `$HOME/outputs-NUMBER` | `/safe_outputs/` | `NUMBER` is a randomly-generated number, for example `outputs-3320888`. This directory is created in your home directory on the back-end. The directory persists after the job which created the container ends. |
+| `$HOME/scratch-NUMBER` | `/scratch/` | `NUMBER` is the same as that created for `outputs-NUMBER`, for example `scratch-3320888`. This directory is also created in your home directory on the back-end. This directory exists for the duration of the job which created the container and is then **deleted** when the job which created the container ends. |
+
+Together, these mounts provides a means for data, configuration files and protoype scripts and code to be shared between the back-end on which the container is running and the environment within the container itself. Creating or editing a file within any of these directories on the back-end means that the changes will be available within the container, and vice-versa.
+
+You can interact with your project's `/safe_data/` subdirectory on the back-end, by logging into the back-end, see [Log into back-ends](./ssh.md).
+
+When using a back-end where your home directory is common to both the Open OnDemand host and the back-end, then you can interact with both `outputs-NUMBER` and `scratch-NUMBER` (and `$HOME/safe_data/`, if applicable) via the [File Manager](./files.md) and/or by logging into the back-end, see [Log into back-ends](./ssh.md).
+
+When using a back-end where your home directory is **not** common to both the Open OnDemand host and the back-end, then you can interact with `/safe_data/PROJECT_SUBDIRECTORY` (or `$HOME/safe_data/`, if applicable), `outputs-NUMBER` and `scratch-NUMBER` by logging into the back-end, see [Log into back-ends](./ssh.md).
+
+!!! Info
+
+    `/safe_data/PROJECT_SUBDIRECTORY` is **not** available on TRE-level back-ends (e.g., the Superdome Flex). For these, you will need to stage your data to the TRE-level back-end following your project- and safe haven-specific processes for the use of TRE-level services.
+
+### Troubleshooting: 'Cannot open project data: /safe_data/cannot_determine_project_from_groups'
```

Comment:
We should add that the user must contact their RC if they encounter this error.

Reply:
Updated `open-ondemand/containers.md` to direct users to their Research Coordinator if a project-specific `/safe_data/` folder can't be identified.